servers/zms/src/main/java/com/yahoo/athenz/zms/store/jdbc/JDBCConnection.java

```diff
@@ -4130,6 +4157,19 @@ private boolean updateRoleMemberNotificationTimestamp(String server, long timest
         return roles;
     }
 
+    // To avoid firing multiple queries against DB, this function will generate 1 consolidated query for all domains->templates combination
+    public String generateDomainTemplateVersionQuery(Map<String, Integer> templateNameAndLatestVersion) {
```

Comment:
this doesn't need to be public and doesn't look we have any unit tests for this method.

Reply:
this code is covered by unit tests in (JDBCConnectionTest->getDomainFromTemplateName) already. i will mark the method as private.

Reply:
it's not covered by that test. that's just a mock that doesn't look at the query string at all so we never verify that we actually created the correct query string.

Reply:
yes you are right it is just a mock. the code coverage part is taken care of while we run the JDBCconnectiontest but i will add a test to validate the query pattern.

Reply:
Done. i added a test for query validation.